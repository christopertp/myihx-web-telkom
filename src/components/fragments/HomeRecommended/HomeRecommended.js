import React from 'react';
import { useSelector } from 'react-redux';
import useSlider from '../../../hooks/useSlider';
import { remToPx } from '../../../utils/unit';
import styles from './styles.scoped.css';

export default function HomeRecommended() {
  const { isLoading, latestNews } = useSelector(s => s.home);
  const margin = (window.innerWidth < 768) ? remToPx(0.75) : remToPx(1);
  const slider = useSlider(3000, margin);
  return (
    <section className={styles.root}>
      <h4>Recommended for you</h4>
      <ul {...slider}>
        {
          isLoading.h
            ? [...Array.from({ length: 1 }).keys()].map(i => <li className="loading" key={i} />)
            : latestNews.map((i, idx) => <img alt="recommended" key={idx} src={i.image} />)
        }
      </ul>
    </section>
  );
}
