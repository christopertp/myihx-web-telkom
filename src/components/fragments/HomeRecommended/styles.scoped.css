
.root > h4 {
  color: #999999;
  font-size: 0.875rem;
  font-weight: normal;
  letter-spacing: 0.0625rem;
  margin: 1.5rem 0 1rem;
  text-transform: uppercase;
}

.root > ul {
  margin-bottom: 1rem;
  overflow-x: auto;
  padding-bottom: 1rem;
  scroll-snap-type: x mandatory;
  white-space: nowrap;
  width: 100%;
}

.root > ul > li, .root > ul > img {
  border-radius: 0.5rem;
  display: inline-block;
  height: 10rem;
  margin-right: 1rem;
}

.root > ul > li {
  width: 22rem;
}

.root > ul > img {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 1, 0.15);
  scroll-snap-align: start;
}

.root > ul > *:last-child {
  margin-right: 0;
}

@media only screen and (max-width: 48em) {
  .root > h4 { font-size: 0.75rem; }
  .root > ul { margin-bottom: 0.25rem; }
  .root > ul > li {
    height: 8.75rem;
    margin-right: 0.75rem;
    width: 13.75rem;
  }
  .root > ul > img {
    height: 8.75rem;
    margin-right: 0.75rem;
  }
}
