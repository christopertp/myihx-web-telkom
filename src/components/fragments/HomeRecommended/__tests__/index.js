import React from 'react';
import { useSelector } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import HomeRecommended from '../HomeRecommended';

jest.mock('../../../../hooks/useSlider');

jest.mock('../../../../utils/unit', () => ({
  remToPx: v => v,
}));

useSelector.mockImplementation(fn => {
  fn({});
  return { isLoading: { h: false }, latestNews: [{}], };
});

describe('src/components/fragments/HomeRecommended', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<HomeRecommended />);
    expect(tree).toMatchSnapshot();
  });

  test('render loading', () => {
    useSelector.mockImplementationOnce(() => ({ isLoading: { h: true } }));
    window.innerWidth = 200;
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<HomeRecommended />);
    expect(tree).toMatchSnapshot();
  });
});
