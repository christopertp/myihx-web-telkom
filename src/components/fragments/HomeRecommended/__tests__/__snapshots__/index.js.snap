// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fragments/HomeRecommended render 1`] = `
<section>
  <h4>
    Recommended for you
  </h4>
  <ul>
    <img
      alt="recommended"
    />
  </ul>
</section>
`;

exports[`src/components/fragments/HomeRecommended render loading 1`] = `
<section>
  <h4>
    Recommended for you
  </h4>
  <ul>
    <li
      className="loading"
    />
  </ul>
</section>
`;
